app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt

```diff
@@ -198,15 +213,23 @@ private fun SavedHikes(savedHikes: List<StateFlow<Hike>>?, hikesViewModel: Hikes
 
 @Composable
 private fun SavedHikeCardFor(hike: Hike, hikesViewModel: HikesViewModel) {
-  // Ask for elevation if not already available
-  if (!hike.elevation.obtained()) {
+  val elevation: List<Double>?
+  if (hike.elevation is DeferredData.Error) {
+    // Display an empty elevation graph if the data is not available because of an error
+    elevation = emptyList()
+  } else if (!hike.elevation.obtained()) {
+    // Ask for elevation if not already available
     hikesViewModel.retrieveElevationDataFor(hike.id)
+    elevation = null
```

Comment:
Yes, the composable will be recomposed every time the hike changes. In particular, here the elevation is not yet available, and `null` is the value to pass to the graph composable for it to display `Loading data...`, so I pass this after requesting the elevation data to the view model.

And once the elevation data comes back, this will be recomposed.